contracts/rust/adapter/src/jellyfish.rs

```diff
@@ -350,6 +347,17 @@ impl From<ParsedPlonkProof> for Proof<Bn254> {
     }
 }
 
+// NOTE: this extra "middle-man" conversion is the unfortunate result of type in rust binding is auto-generated,
+// and are mutually oblivious of the types in Jellyfish, thus no From can be implemented at either side.
+// Since we cannot implement foreign traits on foreign types, we can only transit via our own `ParsedPlonkProof`.
+impl From<ParsedPlonkProof> for contract_bindings::i_plonk_verifier::PlonkProof {
+    fn from(p: ParsedPlonkProof) -> Self {
+        // parsed_proof is our own defined type, which share exactly the same structure and field types
```

Comment:
Do we have a test to detect and mismatch between the bindings and our own type?

Reply:
Or will something break in an obvious way if there is a mismatch?

Reply:
the tests in `service.rs` where we submit a proof to contract should fail (if there's a mismatch) since are calling this conversion there.
but i agree it might be better to add a dedicated test for this unsafe operation for explicit detection

Reply:
Thanks.